cobra/flux_analysis/summary.py

```diff
@@ -51,106 +197,208 @@ def metabolite_summary(met, solution=None, threshold=0.01, fva=False,
 
-    flux_summary['percent'] = 0
-    total_flux = flux_summary.loc[flux_summary.is_input, "flux"].sum()
-
-    flux_summary.loc[flux_summary.is_input, 'percent'] = \
-        flux_summary.loc[flux_summary.is_input, 'flux'] / total_flux
-    flux_summary.loc[~flux_summary.is_input, 'percent'] = \
-        flux_summary.loc[~flux_summary.is_input, 'flux'] / total_flux
-
-    flux_summary['percent'] = flux_summary.percent.apply(
-        lambda x: '{:.0%}'.format(x))
-
-    if fva is not None:
-        flux_table = tabulate(
-            flux_summary.loc[:, ['percent', 'flux', 'fva_fmt', 'id',
-                                 'reaction']].values, floatfmt=floatfmt,
-            headers=['%', 'FLUX', 'RANGE', 'RXN ID', 'REACTION']).split('\n')
-    else:
-        flux_table = tabulate(
-            flux_summary.loc[:, ['percent', 'flux', 'id', 'reaction']].values,
-            floatfmt=floatfmt, headers=['%', 'FLUX', 'RXN ID', 'REACTION']
-        ).split('\n')
-
-    flux_table_head = flux_table[:2]
-
-    met_tag = "{0} ({1})".format(format_long_string(met.name, 45),
-                                 format_long_string(met.id, 10))
-
-    head = "PRODUCING REACTIONS -- " + met_tag
-    print_(head)
-    print_("-" * len(head))
-    print_('\n'.join(flux_table_head))
-    print_('\n'.join(
-        pd.np.array(flux_table[2:])[flux_summary.is_input.values]))
-
-    print_()
-    print_("CONSUMING REACTIONS -- " + met_tag)
-    print_("-" * len(head))
-    print_('\n'.join(flux_table_head))
-    print_('\n'.join(
-        pd.np.array(flux_table[2:])[~flux_summary.is_input.values]))
-
-
-def model_summary(model, solution=None, threshold=0.01, fva=None, names=False,
-                  floatfmt='.3g'):
-    """
-    Print a summary of the input and output fluxes of the model.
+                fmin = rxn.metabolites[self.met] * \
+                    fva_results.at[rxn.id, "minimum"]
+
+                if abs(fmin) <= abs(fmax):
+                    flux_summary.at[rxn.id, "fmax"] = fmax
+                    flux_summary.at[rxn.id, "fmin"] = fmin
+                else:
+                    # Reverse fluxes.
+                    flux_summary.at[rxn.id, "fmax"] = fmin
+                    flux_summary.at[rxn.id, "fmin"] = fmax
+
+        assert flux_summary["flux"].sum() < 1E-6, "Error in flux balance"
+
+        flux_summary = self._process_flux_dataframe(flux_summary)
+
+        flux_summary['percent'] = 0
+        total_flux = flux_summary.loc[flux_summary.is_input, "flux"].sum()
+
+        flux_summary.loc[flux_summary.is_input, 'percent'] = \
+            flux_summary.loc[flux_summary.is_input, 'flux'] / total_flux
+
+        flux_summary.loc[~flux_summary.is_input, 'percent'] = \
+            flux_summary.loc[~flux_summary.is_input, 'flux'] / total_flux
+
+        flux_summary['percent'] = flux_summary.percent.\
+            apply(lambda x: '{:.0%}'.format(x))
+
+        return flux_summary
+
+    def to_table(self):
```

Comment:
I feel like that should be implemented via `object.__str__` and return a string and not print anything. This would extend to the `print` method automatically that way.

Reply:
I agree on the point of functions not having print statements as side effects. The print statements are remnants of the old code and I kept it as it is. I feel the only downsides of implementing it in `__str__` is that it would bloat the method. We can probably return a string from `to_table` though.